python/foglamp/services/core/api/task.py

```diff
@@ -0,0 +1,244 @@
+from foglamp.common.storage_client.exceptions import StorageServerError
+from foglamp.common import utils
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2018 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_help = """
+    -------------------------------------------------------------------------------
+    | GET POST            | /foglamp/scheduled/task                               |
+    -------------------------------------------------------------------------------
+"""
+
+
+#################################
+#  Task
+#################################
+
+
+async def add_task(request):
+    """
+    Create a new task to run a specific plugin
+
+    :Example:
+     curl -X POST http://localhost:8081/foglamp/scheduled/task -d 
+     '{
+        "name": "North Readings to PI",
+        "plugin": "omf",
+        "type": "north",
+        "schedule_type": 3,
+        "schedule_day": 0,
+        "schedule_time": 0,
+        "schedule_repeat": 30,
+        "schedule_enabled": true,
+        "cmd_params": {
+            "stream_id": "1",
+            "debug_level": "1"
+        }
+     }'
+    """
+
+    try:
+        data = await request.json()
+        if not isinstance(data, dict):
+            raise ValueError('Data payload must be a dictionary')
+
+        name = data.get('name', None)
+        plugin = data.get('plugin', None)
+        task_type = data.get('type', None)
+
+        schedule_type = data.get('schedule_type', None)
+        schedule_day = data.get('schedule_day', None)
+        schedule_time = data.get('schedule_time', None)
+        schedule_repeat = data.get('schedule_repeat', None)
+        enabled = data.get('schedule_enabled', None)
+
+        cmd_params = data.get('cmd_params', None)
+
+        if name is None:
+            raise web.HTTPBadRequest(reason='Missing name property in payload.')
+        if plugin is None:
+            raise web.HTTPBadRequest(reason='Missing plugin property in payload.')
+        if task_type is None:
+            raise web.HTTPBadRequest(reason='Missing type property in payload.')
+        if utils.check_reserved(name) is False:
+            raise web.HTTPBadRequest(reason='Invalid name property in payload.')
+        if utils.check_reserved(plugin) is False:
+            raise web.HTTPBadRequest(reason='Invalid plugin property in payload.')
+        if task_type not in ['south', 'north']:
+            raise web.HTTPBadRequest(reason='Only north and south types are supported.')
+        if cmd_params is not None:
+            if not isinstance(cmd_params, dict):
+                raise web.HTTPBadRequest(reason='cmd_params must be a dict.')
+
+        if schedule_type is None:
+            raise web.HTTPBadRequest(reason='Schedule Type is mandatory')
+        if not isinstance(schedule_type, int) and not schedule_type.isdigit():
+            raise web.HTTPBadRequest(reason='Error in type: {}'.format(schedule_type))
```

Comment:
Schedule type error OR `Error in schedule type` as in payload type is for task type